packages/apps/plugins/plugin-graph/src/graph/types.ts

```diff
@@ -54,13 +152,46 @@ export namespace Graph {
   };
 
   export type Action<TProperties extends { [key: string]: any } = { [key: string]: any }> = {
+    /**
+     * Locally unique identifier for the action.
+     */
     id: string;
+
+    /**
+     * Label to be used when displaying the node.
+     *
+     * @example 'My Action'
+     * @example ['unknown action label, { ns: 'example-plugin' }]
+     */
     label: string | [string, { ns: string; count?: number }];
+
+    /**
+     * Icon to be used when displaying the node.
+     */
     icon?: FC<IconProps>;
+
+    /**
+     * Intent(s) to be invoked when the action is invoked.
+     */
     intent?: Intent | Intent[];
+
+    /**
+     * Properties of the node relevant to displaying the node.
+     *
+     * @example { index: 'a1' }
+     */
     properties: TProperties;
+
+    /**
+     * Sub-actions of the node stored by their id.
+     */
     actionsMap: { [key: string]: Action };
```

Comment:
The getter is a util, it's not stored in both forms. Perhaps relevant is this info about deepsignal https://github.com/luisherranz/deepsignal#get-prop---. By using getters because the graph is a deepsignal, both are reactive even though the state is only stored in one place.